clfd/core.py

```diff
@@ -112,7 +112,15 @@ def from_psrchive(cls, archive):
             DataCube instance wrapping Stokes I data.
         """
         if type(archive) == str:
-            archive = psrchive.Archive_load(archive)
+            if hasattr(psrchive, "Archive_load"):
+                archive = psrchive.Archive_load(archive)
+            else:
+                try:
+                    archive = psrchive.Archive.load(archive)
+                except AttributeError:
+                    raise
```

Comment:
There is no point in catching an exception to raise it immediately again. This try/except block is equivalent to:

```python
archive = psrchive.Archive.load(archive)
```
That would be preferable.

Reply:
Also, I don't think there's any point in trying to pre-empt future psrchive API breakages, it took 5 years to encounter one, so we'll just cross that bridge if we come to it again :smile:

Reply:
Ah yes, that's a good point, I hadn't fully mapped that in my head. 

On the second point, do you mean there's no point having both "checks" in there? Given that there will definitely be places where the older `psrchive.Archive_load` I suspect having both is a good idea? Which I suppose that would mean one could implement it as
```python
try:
    archive = psrchive.Archive_load(archive)
except AttributeError:
    archive = psrchive.Archive.load(archive)
```
and then if that still breaks, you actually want the program to terminate anyway. (Very much the Python idiomatic "ask for forgiveness rather than permission" approach.)

Reply:
Yeah that's good as well.